java/openmldb-import/src/main/java/com/_4paradigm/openmldb/importer/InsertImporter.java

```diff
@@ -0,0 +1,148 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com._4paradigm.openmldb.importer;
+
+import com._4paradigm.openmldb.DataType;
+import com._4paradigm.openmldb.SQLInsertRow;
+import com._4paradigm.openmldb.Schema;
+import com._4paradigm.openmldb.sdk.SqlExecutor;
+import org.apache.commons.csv.CSVRecord;
+import org.apache.commons.lang3.tuple.Pair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.List;
+
+class InsertImporter implements Runnable {
+    private static final Logger logger = LoggerFactory.getLogger(InsertImporter.class);
+    private final SqlExecutor router;
+    private final String dbName;
+    private final String tableName;
+    private final List<CSVRecord> records;
+    private final Pair<Integer, Integer> range;
+
+    public InsertImporter(SqlExecutor router, String dbName, String tableName, List<CSVRecord> records, Pair<Integer, Integer> range) {
+        this.router = router;
+        this.dbName = dbName;
+        this.tableName = tableName;
+        this.records = records;
+        this.range = range;
+    }
+
+    @Override
+    public void run() {
+        logger.info("Thread {} insert range [{},{}) ", Thread.currentThread().getId(), range.getLeft(), range.getRight());
+        if (records.isEmpty()) {
+            logger.warn("records is empty");
+            return;
+        }
+        // Use one record to generate insert place holder
+        StringBuilder builder = new StringBuilder("insert into " + tableName + " values(");
+        CSVRecord peekRecord = records.get(0);
+        for (int i = 0; i < peekRecord.size(); ++i) {
+            builder.append((i == 0) ? "?" : ",?");
+        }
+        builder.append(");");
+        String insertPlaceHolder = builder.toString();
+        SQLInsertRow insertRow = router.getInsertRow(dbName, insertPlaceHolder);
+        if (insertRow == null) {
+            logger.warn("get insert row failed");
+            return;
+        }
+        Schema schema = insertRow.GetSchema();
+
+        // TODO(hw): record.getParser().getHeaderMap() check schema? In future, we may read multi files, so check schema in each worker.
+        // What if the header of record is missing?
+        List<String> stringCols = getStringColumnsFromSchema(schema);
+        for (int i = range.getLeft(); i < range.getRight(); i++) {
+            // insert placeholder
+            SQLInsertRow row = router.getInsertRow(dbName, insertPlaceHolder);
```

Comment:
SQLInsertRow is c++ object, need delete explicitly after used?

Reply:
java own it. `this(sql_router_sdkJNI.new_SQLInsertRow(..., true);`.
The last arg is `swigCMemOwn`.
finalize() will delete.